cmd/snap-gdb-shim/snap-gdb-shim.c

```diff
@@ -29,6 +29,13 @@ int main(int argc, char **argv)
 			printf("-%s-\n", argv[i]);
 		}
 	}
+        // check if we run as SUDO and if so switch to a normal user
+        const char *sudo_uid_env = getenv("SUDO_UID");
+        if (sudo_uid_env != NULL) {
+           int sudo_uid = atoi(sudo_uid_env);
+           if (sudo_uid != 0)
+              setuid(sudo_uid);
```

Comment:
You'll want to check return codes as well. Granted, failure to drop when the snap command is called via sudo is not catastrophic since the application is running under confinement, but best practice is, well, best. :)